analyzer/evmnfts/evm_nfts.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/oasisprotocol/nexus/storage"
+	"github.com/oasisprotocol/nexus/storage/client"
+	"github.com/oasisprotocol/nexus/storage/oasis/nodeapi"
+)
+
+const (
+	evmNFTsAnalyzerPrefix = "evm_nfts_"
+)
+
+type processor struct {
+	runtime    common.Runtime
+	source     nodeapi.RuntimeApiLite
+	ipfsClient ipfsclient.Client
+	target     storage.TargetStorage
+	logger     *log.Logger
+}
+
+var _ item.ItemProcessor[*StaleNFT] = (*processor)(nil)
+
+func NewAnalyzer(
+	runtime common.Runtime,
+	cfg config.ItemBasedAnalyzerConfig,
+	sourceClient nodeapi.RuntimeApiLite,
+	ipfsClient ipfsclient.Client,
+	target storage.TargetStorage,
+	logger *log.Logger,
+) (analyzer.Analyzer, error) {
+	logger = logger.With("analyzer", evmNFTsAnalyzerPrefix+runtime)
+	p := &processor{
+		runtime:    runtime,
+		source:     sourceClient,
+		ipfsClient: ipfsClient,
+		target:     target,
+		logger:     logger,
+	}
+	return item.NewAnalyzer[*StaleNFT](
+		evmNFTsAnalyzerPrefix+string(runtime),
+		cfg,
+		p,
+		target,
+		logger,
+	)
+}
+
+type StaleNFT struct {
+	Addr                  string
+	ID                    *big.Int
+	Type                  *evm.EVMTokenType
+	AddrContextIdentifier string
+	AddrContextVersion    int
+	AddrData              []byte
+	DownloadRound         uint64
+}
+
+func (p *processor) GetItems(ctx context.Context, limit uint64) ([]*StaleNFT, error) {
+	var staleNFTs []*StaleNFT
+	rows, err := p.target.Query(ctx, queries.RuntimeEVMNFTAnalysisStale, p.runtime, limit)
+	if err != nil {
+		return nil, fmt.Errorf("querying discovered NFTs: %w", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var staleNFT StaleNFT
+		var idC common.BigInt
+		if err = rows.Scan(
+			&staleNFT.Addr,
+			&idC,
+			&staleNFT.Type,
+			&staleNFT.AddrContextIdentifier,
+			&staleNFT.AddrContextVersion,
+			&staleNFT.AddrData,
+			&staleNFT.DownloadRound,
+		); err != nil {
+			return nil, fmt.Errorf("scanning discovered nft: %w", err)
+		}
+		staleNFT.ID = &idC.Int
+		staleNFTs = append(staleNFTs, &staleNFT)
+	}
+	return staleNFTs, nil
+}
+
+func (p *processor) ProcessItem(ctx context.Context, batch *storage.QueryBatch, staleNFT *StaleNFT) error {
+	p.logger.Info("downloading", "stale_nft", staleNFT)
+	tokenEthAddr, err := client.EVMEthAddrFromPreimage(staleNFT.AddrContextIdentifier, staleNFT.AddrContextVersion, staleNFT.AddrData)
+	if err != nil {
+		return fmt.Errorf("token address: %w", err)
+	}
+	nftData, err := evm.EVMDownloadNewNFT(
+		ctx,
+		p.logger,
+		p.source,
+		p.ipfsClient,
+		staleNFT.DownloadRound,
+		tokenEthAddr,
+		*staleNFT.Type,
+		staleNFT.ID,
+	)
+	if err != nil {
+		return fmt.Errorf("downloading NFT %s %v: %w", staleNFT.Addr, staleNFT.ID, err)
+	}
+	batch.Queue(queries.RuntimeEVMNFTUpdate, p.runtime, staleNFT.Addr, staleNFT.ID, nftData.MetadataURI, nftData.MetadataAccessed, nftData.Name, nftData.Description, nftData.Image)
+	batch.Queue(queries.RuntimeEVMNFTAnalysisUpdate, p.runtime, staleNFT.Addr, staleNFT.ID, staleNFT.DownloadRound)
```

Comment:
this can probably be a single query now?

Reply:
oh yeah